kafka/producer.go

```diff
@@ -130,7 +110,9 @@ func (cfg *ProducerConfig) finalize() error {
 	return errors.Join(errs...)
 }
 
-// Producer is a model.BatchProcessor that publishes events to Kafka.
+var _ apmqueue.Producer = &Producer{}
```

Comment:
nit (feel free to ignore): I assume this it to check if `apmqueue.Producer` is implemented by `*Producer`. Maybe we can do something like below:

```suggestion
var _ apmqueue.Producer = (*Producer)(nil)
```